cmd/root.go

```diff
@@ -63,8 +63,11 @@ func NewWithViper(vp *viper.Viper) *cobra.Command {
 			vp.SetConfigFile(cfg)
 		}
 		// if a config file is found, read it in.
-		if err := vp.ReadInConfig(); err == nil && vp.GetBool(config.KeyDebug) {
-			fmt.Fprintln(rootCmd.ErrOrStderr(), "Using config file:", vp.ConfigFileUsed())
+		err := vp.ReadInConfig()
+		// initialize the logger after all the config parameters get loaded to viper.
+		logger.Initialize(vp)
```

Comment:
this looks a bit weird but somehow i need to wait until all the config values get loaded before initializing the logger 😐